MatrixSDK/JSONModels/MXEvent.m

```diff
@@ -472,12 +486,30 @@ - (BOOL)isReplyEvent
 
 - (BOOL)isVoiceMessage
 {
-    NSString *msgtype = self.content[@"msgtype"];
+    NSString *msgtype = self.content[kMXMessageTypeKey];
     return [msgtype isEqualToString:kMXMessageTypeAudio] && (self.content[kMXMessageContentKeyVoiceMessage] ||
                                                              self.content[kMXMessageContentKeyVoiceMessageMSC2516] ||
                                                              self.content[kMXMessageContentKeyVoiceMessageMSC3245]);
 }
 
+- (MXEventContentLocation *)location
```

Comment:
Normally we are more adding `- (BOOL)isLocation` kind of method on `MXEvent` rather than building another kind of event from the current one.
Otherwise is it possible to handle:
```
    NSDictionary *locationContent = self.content[kMXMessageContentKeyExtensibleLocationMSC3488];
     if (locationContent == nil) {
         locationContent = self.content[kMXMessageContentKeyExtensibleLocation];
     }
```
directly in `MXEventContentLocation modelFromJSON:`?

Reply:
I was unsure if I should pass the whole event payload in there but you're right, it does make things cleaner 👍